app/src/main/java/com/runnect/runnect/presentation/detail/CourseDetailViewModel.kt

```diff
@@ -1,172 +1,169 @@
-    private val _editState = MutableLiveData<UiState>()
+    val isValidTitle: LiveData<Boolean> = _title.map { it.isNotBlank() }
+    val isValidDescription: LiveData<Boolean> = _description.map { it.isNotBlank() }
 
-    private var _editMode = MutableLiveData(false)
-    val editMode: LiveData<Boolean>
-        get() = _editMode
+    // 액티비티에서 참조하는 변수
+    private var _currentScreenMode: ScreenMode = ScreenMode.ReadOnlyMode
+    val currentScreenMode get() = _currentScreenMode
 
-    private var _courseDetail = CourseDetailDTO(
-        "CSPR0", "1", "1", 1, "1", "1", "1", 1, "1", false, "1", listOf(listOf(1.1))
-    )
+    private var savedContents = CourseDetailContents("", "")
 
-    val courseDetail: CourseDetailDTO
-        get() = _courseDetail
+    fun updateCourseTitle(title: String) {
+        _title.value = title
+    }
 
-    val errorMessage = MutableLiveData<String>()
+    fun updateCourseDescription(desc: String) {
+        _description.value = desc
+    }
+
+    fun saveCurrentContents() {
+        savedContents = CourseDetailContents(title, description)
+    }
+
+    fun restoreOriginalContents() {
+        _title.value = savedContents.title
+        _description.value = savedContents.description
+    }
 
-    fun convertMode() {
-        _editMode.value = !_editMode.value!!
+    fun updateCurrentScreenMode(mode: ScreenMode) {
+        _currentScreenMode = mode
     }
 
     fun getCourseDetail(courseId: Int) {
         viewModelScope.launch {
-            runCatching {
-                _courseDetailState.value = UiState.Loading
-                courseRepository.getCourseDetail(
-                    publicCourseId = courseId
-                )
-            }.onSuccess {
-                imageUrl.value = it.image
-                title.value = it.title
-                description.value = it.description
-
-                _courseDetail = it
-                stampId.value = it.stampId
-                mapImage.value = it.image
-                editTitle.value = it.title
-                editContent.value = it.description
-                distance.value = it.distance
-                departure.value = it.departure
-                _courseDetailState.value = UiState.Success
-            }.onFailure {
-                _courseDetailState.value = UiState.Failure
-                errorMessage.value = it.message
-
-                if(errorMessage.value == "HTTP 401 Unauthorized"){ //딥링크 접속 && 미로그인의 경우
-                    isDeepLinkLogin.value = false
+            _courseGetState.value = UiStateV2.Loading
+
+            courseRepository.getCourseDetail(
+                publicCourseId = courseId
+            ).onSuccess { response ->
+                _courseGetState.value = UiStateV2.Success(response)
+                Timber.d("[SUCCESS] GET COURSE DETAIL")
+            }.onFailure { t ->
+                _courseGetState.value = UiStateV2.Failure(t.message.toString())
+
+                if (t is HttpException) {
```

Comment:
여기서 발생할 수 있는 HttpException이란 어떤 것일까요?

Reply:
말그대로 HTTP 서버통신 과정에서 발생할 수 있는 예외를 처리하는 클래스입니다. 

```java

/** Exception for an unexpected, non-2xx HTTP response. */
public class HttpException extends RuntimeException {
  private static String getMessage(Response<?> response) {
    Objects.requireNonNull(response, "response == null");
    return "HTTP " + response.code() + " " + response.message();
  }

  private final int code;
  private final String message;
  private final transient Response<?> response;

  public HttpException(Response<?> response) {
    super(getMessage(response));
    this.code = response.code();
    this.message = response.message();
    this.response = response;
  }

  /** HTTP status code. */
  public int code() {
    return code;
  }

  /** HTTP status message. */
  public String message() {
    return message;
  }

  /** The full HTTP response. This may be null if the exception was serialized. */
  public @Nullable Response<?> response() {
    return response;
  }
}
```

이 클래스를 이용하면 위와 같이 code, message, response 정보들을 참조할 수 있어서 사용했습니다!

Reply:
이렇게 if문으로 (t is HttpException)를 안 만들어줘도 fail 시 message 등의 값들을 참조할 수 있지 않나요?! 

저는 그동안 이렇게 HttpException 조건문 블록을 열어주지 않았는데도 fail 시 어떤 상황 때문에 일어난 건지 log를 찍어볼 수 있었거든요. 제가 잘 모르는 것일 수도(그동안 잘못알고 있던 것일 수도) 있는데 이렇게 HttpException 조건문 블록을 열어주는 것이 어떤 의도인지, 어떤 효용을 얻을 수 있는지 궁금합니다.

Reply:
그런데 현재 HttpLoggingInterceptor 이용해서 서버통신과 관련된 자세한 로그 정보를 확인할 수 있기 때문에, Timber로 로그를 찍는 건 불필요하다고 판단하여 관련 코드를 삭제했습니다!